src/System.Net.HttpListener/tests/HttpListenerContextTests.cs

```diff
@@ -27,6 +30,11 @@ public void Dispose()
         {
             Factory.Dispose();
             Socket.Dispose();
+
+            if (ClientConnectTask?.IsCompleted == true)
+            {
+                ClientConnectTask?.Dispose();
```

Comment:
You don't need to dispose of the task.  Please remove this and the corresponding changes in GetWebSocketContext.